src/server/session.ts

```diff
@@ -607,27 +622,25 @@ namespace ts.server {
                                 NewLineCharacter: "\n",
                                 ConvertTabsToSpaces: formatOptions.ConvertTabsToSpaces,
                             };
-                            var indentPosition =
-                                compilerService.languageService.getIndentationAtPosition(file, position, editorOptions);
+                            var preferredIndent = compilerService.languageService.getIndentationAtPosition(file, position, editorOptions);
+                            var hasIndent = 0;
                             for (var i = 0, len = lineText.length; i < len; i++) {

```

Comment:
we should already have something that is kind of doing the same thing [findFirstNonWhitespaceCharacterAndColumn](https://github.com/Microsoft/TypeScript/blob/master/src/services/formatting/smartIndenter.ts#L375). Can you check if it can be reused here?